pyclam/manifold.py

```diff
@@ -486,33 +487,47 @@ def clear_cache(self) -> None:
         self.clusters = {c: None for c in self.clusters.keys()}
         return
 
-    def random_walk(self, steps: int = 5, walks: int = 1) -> Dict[Cluster, int]:
-        """ Performs a random walk, returning a modified graph instance.
-
-        :param int steps: number of steps per walk
-        :param int walks: number of walks to perform
-        :returns a Dict of cluster names to visit counts
-        """
-        clusters = list(self.clusters.keys())
-        results = {c: list() for c in clusters}
+    def _random_walks(self, clusters: List[Cluster], steps: int) -> Dict[Cluster, int]:
```

Comment:
Can this functionality be moved into `random_walks`? Let's keep namespace clean if we can since python doesn't provide true privates